### PR TITLE
feat: Add ReplicatorError and map it to JS layer.

### PR DESCRIPTION
**Pull request by fmarek-kindred**

### CODE REVIEW

File: packages/talos_cohort_replicator/src/talos_cohort_replicator.rs [hunk 1]
```diff
@@ -57,15 +57,18 @@ pub struct CohortReplicatorConfig {
...
 }
 
-async fn flatten_service_result<T>(handle: JoinHandle<Result<T, ServiceError>>) -> Result<T, ServiceError> {
+async fn flatten_service_result<T>(handle: JoinHandle<Result<T, ReplicatorError>>) -> Result<T, ReplicatorError> {
     match handle.await {
         Ok(Ok(result)) => Ok(result),
         Ok(Err(err)) => Err(err),
-        Err(err) => Err(ServiceError {
-            reason: format!("handling failed with error={err:?}"),
+        Err(err) => Err(ReplicatorError {
```
Comment (fmarek-kindred) [inline packages/talos_cohort_replicator/src/talos_cohort_replicator.rs:64]: Don't know what needs to happen for this to fail. Usually if service fails to start the error is delivered via `Ok(Err)` branch.

File: cohort_banking_replicator_js/package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
   },
   "dependencies": {
     "@types/pg": "^8.10.2",
-    "cohort_sdk_js": "file:../packages/cohort_sdk_js",
+    "cohort_sdk_client": "file:../cohort_sdk_client",
     "nodejs-dashboard": "^0.5.1",
     "pg": "^8.11.3",
     "ts-node": "^10.9.1",
```
Comment (gk-kindred) [inline cohort_banking_replicator_js/package.json:23]: do we need `ts-node`? Since you use `tsc` to transpile `ts` to `js`, I think we can remove this.

**Comment by gk-kindred** [comment]
Two minor comments. Also please could you test `initiator` and `replicator` js version for `node 18` as all teams will soon move to node v18.

**Review by gk-kindred — APPROVED** [review]

File: cohort_banking_replicator_js/package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
   },
   "dependencies": {
     "@types/pg": "^8.10.2",
```
[the remark this answers is not in the document]
Comment (fmarek-kindred) [reply cohort_banking_replicator_js/package.json]: Thanks.

File: cohort_banking_replicator_js/package.json (hunk 2 above)
Comment (fmarek-kindred) [reply cohort_banking_replicator_js/package.json]: You are right this can be removed.

**Comment by fmarek-kindred** [comment]
> Two minor comments. Also please could you test `initiator` and `replicator` js version for `node 18` as all teams will soon move to node v18.

Thanks. I added KDT-91 ticket for this.

**Pull request merged**